integration-tests/standalone-asm.spec.js

```diff
@@ -0,0 +1,302 @@
+        assert.isArray(payload)
+        assert.strictEqual(payload.length, 4)
+
+        assertKeep(payload[3][0])
+      })
+    })
+
+    it('should keep sdk events', async () => {
+      await doWarmupRequests(proc)
+
+      const url = proc.url + '/login?user=test'
+      return curlAndAssertMessage(agent, url, ({ headers, payload }) => {
+        assert.propertyVal(headers, 'datadog-client-computed-stats', 'yes')
+        assert.isArray(payload)
+        assert.strictEqual(payload.length, 4)
+
+        assertKeep(payload[3][0])
+      })
+    })
+
+    it('should keep custom sdk events', async () => {
+      await doWarmupRequests(proc)
+
+      const url = proc.url + '/sdk'
+      return curlAndAssertMessage(agent, url, ({ headers, payload }) => {
+        assert.propertyVal(headers, 'datadog-client-computed-stats', 'yes')
+        assert.isArray(payload)
+        assert.strictEqual(payload.length, 4)
+
+        assertKeep(payload[3][0])
+      })
+    })
+
+    it('should keep iast events', async () => {
+      await doWarmupRequests(proc)
+
+      const url = proc.url + '/vulnerableHash'
+      return curlAndAssertMessage(agent, url, ({ headers, payload }) => {
+        assert.propertyVal(headers, 'datadog-client-computed-stats', 'yes')
+        assert.isArray(payload)
+        assert.strictEqual(payload.length, 4)
+
+        const expressReq4 = payload[3][0]
+        assertKeep(expressReq4)
+        assert.property(expressReq4.meta, '_dd.iast.json')
+        assert.propertyVal(expressReq4.metrics, '_dd.iast.enabled', 1)
+      })
+    })
+
+    describe('propagation', () => {
+      let proc2
+      let port2
+
+      beforeEach(async () => {
+        const execArgv = []
+
+        proc2 = await spawnProc(startupTestFile, { cwd, env, execArgv })
+
+        port2 = parseInt(proc2.url.substring(proc2.url.lastIndexOf(':') + 1), 10)
+      })
+
+      afterEach(async () => {
+        proc2.kill()
+      })
+
+      // proc/drop-and-call-sdk:
+      // after setting a manual.drop calls to downstream proc2/sdk which triggers an appsec event
+      it('should keep trace even if parent prio is -1 but there is an event in the local trace', async () => {
+        await doWarmupRequests(proc2)
+
+        const url = `${proc.url}/propagation-after-drop-and-call-sdk?port=${port2}`
+        return curlAndAssertMessage(agent, url, ({ headers, payload }) => {
+          assert.propertyVal(headers, 'datadog-client-computed-stats', 'yes')
+          assert.isArray(payload)
+
+          const innerReq = payload.find(p => p[0].resource === 'GET /sdk')
+          assert.notStrictEqual(innerReq, undefined)
+          assertKeep(innerReq[0])
+        }, undefined, undefined, true)
+      })
+
+      // proc/propagation-with-event triggers an appsec ev and calls downstream proc2/down with no event
+      it('should keep if parent trace is (prio:2, _dd.p.appsec:1) but there is no ev in the local trace', async () => {
```

Comment:
you are not doing a warmup of `proc2`, doesn't it means that in `proc2` all the requests are the first request and should be kept always?

Reply:
oops, fixed. now there is a warmup for both processes.